warehouse/query-core/src/main/java/datawave/query/planner/FederatedQueryPlanner.java

```diff
@@ -0,0 +1,329 @@
+            Multimap<String,Type<?>> fieldToDatatypeMap = FetchDataTypesVisitor.fetchDataTypes(metadataHelper,
+                            ((ShardQueryConfiguration) config).getDatatypeFilter(), queryTree, false);
+            
+            try {
+                calculateFieldIndexHoles(metadataHelper, fieldToDatatypeMap, (ShardQueryConfiguration) config);
+            } catch (TableNotFoundException e) {
+                log.error("metadata table was not found " + e.getMessage());
+            }
+            
+            List<FieldIndexHole> fieldIndexHoles = ((ShardQueryConfiguration) config).getFieldIndexHoles();
+            List<ValueIndexHole> valueIndexHoles = ((ShardQueryConfiguration) config).getValueIndexHoles();
+            holeDates = generateStartAndEndDates((ShardQueryConfiguration) config);
+            if ((valueIndexHoles == null && fieldIndexHoles == null) || (valueIndexHoles.size() == 0 && fieldIndexHoles.size() == 0) || holeDates.size() == 0) {
+                results = super.process(config, query, settings, scannerFactory);
+                returnQueryData.addDelegate(results);
+                return returnQueryData;
+            }
+            
+            Boolean firstIteration = true;
+            Date startDate, endDate;
+            
+            for (Iterator<YearMonthDay> it = holeDates.iterator(); it.hasNext();) {
+                if (firstIteration) {
+                    firstIteration = false;
+                    startDate = originalStartDate;
+                    if (it.hasNext()) {
+                        endDate = DateHelper.parse(it.next().getYyyymmdd());
+                    } else
+                        endDate = originalEndDate;
+                } else {
+                    startDate = DateHelper.parse(it.next().getYyyymmdd());
+                    if (it.hasNext())
+                        endDate = DateHelper.parse(it.next().getYyyymmdd());
+                    else {
+                        endDate = originalEndDate;
+                    }
+                }
+                
+                results = getQueryData((ShardQueryConfiguration) config, query, settings, scannerFactory, startDate, endDate);
+                returnQueryData.addDelegate(results);
+                
+            }
+            
+        }
+        
+        return returnQueryData;
+    }
+    
+    private CloseableIterable<QueryData> getQueryData(ShardQueryConfiguration config, String query, Query settings, ScannerFactory scannerFactory,
+                    Date startDate, Date endDate) throws DatawaveQueryException {
+        log.debug("getQueryData in the FederatedQueryPlanner is called ");
+        CloseableIterable<QueryData> queryData;
+        ShardQueryConfiguration tempConfig = new ShardQueryConfiguration(config);
+        tempConfig.setBeginDate(startDate);
+        tempConfig.setEndDate(endDate);
+        // TODO: I think it is unnecessary to clone the DefaultQueryPlanner but I think it was requested.
```

Comment:
You maybe correct here as long as the results of tempPlanner.process do not depend on any members of the tempPlanner.... need to verify that.

Reply:
The code that you have to trace through is so long and in so many classes that I just wanted to be safe.